zerver/test_external.py

```diff
@@ -225,3 +226,16 @@ def test_change_user(self):
         result = self.client.post('/json/users/me/android_gcm_reg_id', {'token':token})
         self.assert_json_success(result)
 
+class SanitizeNameTests(TestCase):
+    def test_file_name(self):
+        fileName=sanitize_name('test.txt')
+        hiddenFile=sanitize_name('.hidden')
+        hiddenFileWithExtension=sanitize_name('.hidden.txt')
+        tarball=sanitize_name('tarball.tar.gz')
+        hiddenTarball=sanitize_name('.hiddentarball.tar.gz')
+        self.assertEquals(fileName, 'test.txt')
+        self.assertEquals(hiddenFile, '.hidden')
+        self.assertEquals(hiddenFileWithExtension, '.hidden.txt')
+        self.assertEquals(tarball, 'tarball.tar.gz')
+        self.assertEquals(hiddenTarball, '.hiddentarball.tar.gz')
+
```

Comment:
(It's probably worth your reading a little bit about the dangers of allowing a user to create files with arbitrary names as you work on this patch, since your initial version would have introduced a security bug into Zulip)